tycho-core/src/main/java/org/eclipse/tycho/core/ee/ListSystemPackages.java

```diff
@@ -19,17 +21,27 @@
 public class ListSystemPackages {
 
     public static void main(String[] args) {
+        Version version = Runtime.version();
+        try {
+            System.out.println(version.feature());
+        } catch (NoSuchMethodError e) {
+            System.out.println("9");
+        }
         getCurrentJREPackages().forEach(System.out::println);
     }
 
     public static Set<String> getCurrentJREPackages() {
         return ModuleLayer.boot().modules().stream().map(Module::getDescriptor) //
                 .flatMap(desc -> desc.isAutomatic() ? //
                         desc.packages().stream() : //
-                        desc.exports().stream()
-                                .filter(Predicate.not(java.lang.module.ModuleDescriptor.Exports::isQualified))
+                        desc.exports().stream().filter(not(java.lang.module.ModuleDescriptor.Exports::isQualified))
                                 .map(java.lang.module.ModuleDescriptor.Exports::source) //
                 ).collect(Collectors.toSet());
     }
 
+    private static Predicate<? super Exports> not(Predicate<Exports> predicate) {
+        // java 9 compatible polyfill
+        return x -> !predicate.test(x);
+    }
+
```

Comment:
Can you please extract this into a dedicated patch?

Reply:
https://github.com/eclipse/tycho/pull/996